info-compiler/src/main/java/com/google/sps/infocompiler/InfoCompiler.java

```diff
@@ -261,18 +270,46 @@ void storeElectionContestInDatabase(String electionQueryId, JsonObject contest)
     datastore.update(electionEntity);
   }
 
-  // @TODO [Find incumbency.]
+  private boolean checkIncumbency(String candidateName, String position, String division) {
+    try {
+      String queryUrl = String.format("%s&ocdDivisionId=%s", REPRENTATIVE_QUERY_URL, division);
+      JsonObject representatives = queryCivicInformation(queryUrl);
+      JsonArray offices = representatives.getAsJsonArray("offices");
+      int i = 0;
+      for (JsonElement eachOffice : offices){
+        JsonObject office = (JsonObject) eachOffice;
+        String officeName = office.get("name").getAsString();
+        if (position.equals(officeName)){
+          break;
+        }
+        i++;
+      }
+      JsonObject officials = representatives.getAsJsonObject("officials");
+      String incumbent = (officials.get("name").getAsString());
```

Comment:
`officials` is a `JsonArray`. There needs to be a `officials.get(i)` for extracting the particular official who's holding the position we are interested in.

Reply:
Is this part of your previous explanation or is this part of the review?

Reply:
This is talking about the same thing as the above response to Kelly: https://github.com/googleinterns/voter-central/pull/27#discussion_r463633940